common4j/src/main/com/microsoft/identity/common/java/telemetry/adapter/BrokerTelemetryAdapter.java

```diff
@@ -21,16 +21,93 @@
 // THE SOFTWARE.
 package com.microsoft.identity.common.java.telemetry.adapter;
 
+import com.microsoft.identity.common.java.telemetry.TelemetryEventStrings;
 import com.microsoft.identity.common.java.telemetry.observers.IBrokerTelemetryObserver;
+import com.microsoft.identity.common.java.util.StringUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
 import lombok.NonNull;
 
 /**
  * Adapter to aggregate events for broker telemetry. We make use of the {@link TelemetryAggregationAdapter} which
  * aggregates the events into MATS format.
+ * <p>
+ * We then send out the error events as single events since the {@link TelemetryAggregationAdapter} will pick the last error event
+ * and add combine it with the aggregated event. This means that the event sent by {@link TelemetryAggregationAdapter} only contains
+ * the last error that was dispatched.
  */
 public class BrokerTelemetryAdapter extends TelemetryAggregationAdapter {
     public BrokerTelemetryAdapter(@NonNull IBrokerTelemetryObserver observer) {
         super(observer);
     }
+
+    @Override
+    public void process(@NonNull List<Map<String, String>> rawData) {
```

Comment:
Can you check where (on which thread) does all the processing happen? I'm thinking this probably doesn't need to happen on the same thread that was doing the AT / ATS request and this processing can be done on a separate thread. We don't have to solve the problem right now but something to look at to come up with a plan to improve this going forward.